### PR TITLE
[ELY-2752] Ensure it's possible to make use of a custom principal-attribute value for OIDC

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/ELY-2752

### CODE REVIEW

**Review by ivassile — APPROVED** [review]

File: http/oidc/src/main/java/org/wildfly/security/http/oidc/JsonWebToken.java [hunk 1]
```diff
@@ -297,7 +297,14 @@ public String getPrincipalName(OidcClientConfiguration deployment) {
             case NICKNAME:
                 return getNickName();
             default:
-                return getSubject();
+                String claimValue = getClaimValueAsString(attr);
+                if (claimValue != null) {
+                    return claimValue;
+                } else {
+                    // fall back to sub claim
+                    log.principalAttributeClaimDoesNotExist(attr);
```
Comment (darranl) [inline http/oidc/src/main/java/org/wildfly/security/http/oidc/JsonWebToken.java:305]: This looks like it has the potential to become per-request - I am wondering if this should instead be debug information?

Comment (fjuma) [reply http/oidc/src/main/java/org/wildfly/security/http/oidc/JsonWebToken.java:305]: Good catch, I've fixed this, thanks!

**Comment by fjuma** [comment]
Hi @darranl, I've made the suggested update. Did you want to do a quick pass to make sure this looks good to you now?

**Review by darranl — APPROVED** [review]
Looks good, thank you @fjuma

**Pull request merged**